src/corecel/math/ArrayUtils.hh

```diff
@@ -198,6 +204,33 @@ is_soft_orthogonal(Array<T, N> const& x, Array<T, N> const& y)
     return soft_zero(dot_product(x, y));
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Check whether two vectors are approximately collinear.
+ *
+ * This is done by checking if the Gram determinant is soft zero:
+ * \f[
+ * (x \cdot x) (y \cdot y) - (x \cdot y)^2 \approx 0
+ * \f]
+ * This method is advantageous because it doesn't require normalizing inputs.
+ */
+template<class T, size_type N>
+inline CELER_FUNCTION bool
+is_soft_collinear(Array<T, N> const& x, Array<T, N> const& y)
+{
+    SoftZero const soft_zero{SoftEqual{}.rel()};
+    T const xxyy = dot_product(x, x) * dot_product(y, y);
+
+    if (soft_zero(xxyy))
```

Comment:
If `x` and `y` aren't unit vectors, the soft zero test isn't meaningful: I think there should be an implicit documented precondition that these are both unit vectors, and a CELER_EXPECT(!soft_zero(xxyy));` here.

(In the doxygen, use `\pre Both x and y must be approximately unit vectors`)

Reply:
(and I didn't want to use make_unit_vector because that involves a square root)